adamsjr8576/rancid-tomatillos
# App testing

#### What's this PR do?

-This PR creates a Loading component and gives it functionality/css animation for when a user is awaiting the movies fetching to resolve

-This PR also writes tests for the Loading, App, and MoviePage components, tests for the apiCalls, and the movies reducer. 

#### Where should the reviewer start?

Reviewer should probably start with the testing files to make sure they are robust. I would start with the apiCalls and movies reducer as those are fairly straight forward. Next I would look at the App test and make sure that everything that should be tested is covered. Finally, I would review the MoviePage testing and make sure that everything that needs to be tested is covered. 

Note: There are a number of snapshots for the MoviePage and as the findMovieRating method is conditionally rendering html based on the value of its inputs, I opted to let the snapshots account for the testing of that method. 


#### How should this be manually tested?

Npm test or you can run npm start and try refreshing your browser a number of times to try and see the Loading component in action.
